newsroom/agenda/agenda.py

```diff
@@ -315,13 +310,46 @@ def _agenda_query():
     }
 
 
+def get_date_filter_query(filter_name: str):
+    """Get the query for the given filter name."""
+    time_filters = app.config.get("AGENDA_TIME_FILTERS", [])
+    default_filter = next((f for f in time_filters if f.get("default")), None)
+    if not filter_name:
+        return default_filter["query"]
+    if filter_name == "custom_date":
+        return None
+    for time_filter in time_filters:
+        if time_filter["filter"] == filter_name:
+            query = time_filter["query"]
+            return query
+    return None
+
+
 def get_date_filters(args):
     date_range = {}
     offset = int(args.get("timezone_offset", "0"))
-    if args.get("date_from"):
-        date_range["gt"] = get_local_date(args["date_from"], "00:00:00", offset)
-    if args.get("date_to"):
-        date_range["lt"] = get_end_date(args["date_to"], get_local_date(args["date_to"], "23:59:59", offset))
+
+    agenda_filter = get_date_filter_query(args.get("date_filter"))
+    date_from = args.get("date_from")
+
+    if agenda_filter:
+        if date_from:
+            start_date = get_local_date(date_from, "00:00:00", offset)
+            if agenda_filter.startswith("now-"):  # Handle past filters (e.g., last_7_days)
+                date_range["gt"] = get_local_date(agenda_filter, "00:00:00", offset)
+                date_range["lt"] = start_date
+            elif agenda_filter.startswith("now+"):  # Handle future filters (e.g., next_7_days)
+                date_range["gt"] = start_date
+                date_range["lt"] = get_end_date(agenda_filter, start_date)
+            elif agenda_filter.startswith("now"):  # Handle current filters (e.g., this_week)
+                date_range["gt"] = get_local_date(agenda_filter, "00:00:00", offset)
+                date_range["lt"] = get_end_date(agenda_filter, get_local_date(agenda_filter, "23:59:59", offset))
```

Comment:
I wouldn't use that `get_local_date` on those filters, that requires us to have all the options defined in core and if someone will want *next 10 months* it won't work without core changes, can we rather use those queries as they are without converting them to dates?

Reply:
so in this case we have to update the UI grouping because the list grouping is also used this behaviour currently

Reply:
list grouping is working based on event dates, not filter query afaik

Reply:
I see, hm that's probably quite complex then, not sure how the client would know what to display then

Reply:
that's exactly why I didn't implement the behavior you suggested

Reply:
ok so lets keep it simple for now

Reply:
so that client code needs to be updated too right? to support those `now+30d` etc

Reply:
yep